### ISSUE TITLE
Labels with "[#]" makes PD non-responsive

### ISSUE THREAD

**Issue by HenriAugusto** [pure-data/pure-data#417]
Replicating the issue:

1. use a messagebox to set the label for a gui object to anything that contains an **[#]**
2. right-click and try to open the [hslider]'s properties

![pd label bug](https://user-images.githubusercontent.com/11790621/44881460-1738e580-ac86-11e8-8c42-42bd1d794557.png)

PD becomes non-responsive.

**Comment by danomatika** [comment]
Have you tried this with the Pd 0.49 test builds? I think this is fixed already.

I'm try this with 0.49-0test3 and it seems to be working fine, although the right bracket is sometimes dropped when entering.

**Comment by HenriAugusto** [comment]
I've just did some testing and actually the problem appears to be when use using anything inside square brackets.

Just created a Canvas and set it's label to [test]. Nothing will happend. But when i right-click and try to go to properties i get TCL error on the console

~~~
(Tcl) INVALID COMMAND NAME: invalid command name "test"
    while executing
"test"
    ("uplevel" body line 2)
    invoked from within
"uplevel #0 $docmds"
~~~

Probably because TCL interprets stuff inside square brackets as commands. But I've noticed that even if i tried to set the label to \[test\] i would still get the error when going to the properties

**Comment by HenriAugusto** [comment]
Also if a create a canvas and set it's label to **"["** the next time i try to open it's properties then pd becomes unresponsive.

**Comment by porres** [comment]
@HenriAugusto , did you notice "#" becomes "$"? It's a separate issue, I guess, but it's something that happens...

**Comment by electrickery** [comment]
With 0.50.2 there is no problem with a # in the iem_gui labels. 
Yes, it is mutated into a $. This might be related to a hack which makes it possible to use $ in messages, at the cost of not being able to use #. 
The non-responsiveness doesn't occur (Xubuntu 18.04 64-bit).

**Comment by porres** [comment]
possible duplicate? => https://github.com/pure-data/pure-data/issues/880

**Comment by HenriAugusto** [comment]
I figure we should escape square brackets to prevent [command evaluation](https://www.tcl.tk/man/tcl8.5/tutorial/Tcl5.html) when sending the message do the GUI.

**Comment by umlaeute** [comment]
@HenriAugusto problems with square brackets are tracked in #880, which is **not** a duplicate of this ticket. (and #880 also outlines how to do it)

**Comment by porres** [comment]
> @HenriAugusto problems with square brackets are tracked in #880, which is not a duplicate of this ticket. (and #880 also outlines how to do it)

I don't understand why this is not a duplicate of https://github.com/pure-data/pure-data/issues/880

There's a fix for #880 now, see https://github.com/pure-data/pure-data/pull/948

I tried it, and it also fixes this issue here...

what is different in this issue is that we're already talking about "#"/"$" conversion, which is also a duplicate of https://github.com/pure-data/pure-data/issues/558

Anyway, can we close this issue?
